File: LeetCode_904/Main.py
class Solution(object):
    def totalFruit(self, fruits):
        """
        :type fruits: List[int]
        :rtype: int
        """
        cnt = dict()
        ans = 1
        j = 0
        for i in range(len(fruits)):
            if fruits[i] in cnt:
                cnt[fruits[i]] += 1
            else:
                cnt[fruits[i]] = 1

            while len(cnt) > 2:
                cnt[fruits[j]] -= 1
                if cnt[fruits[j]] == 0:
                    del fruits[i]
                j += 1
            ans = max(ans, i - j + 1)
        return ans

File: LeetCode_904/Main.cpp
#include<bits/stdc++.h>
using namespace std;


class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        map<int, int> cnt;
        int ans = 2;

        for (int i = 0, j = 0; i < fruits.size(); i++) {
            cnt[fruits[j]]++;
            while (cnt.size() > 2) {
                cnt[fruits[j]]--;
                if (cnt[fruits[j]] == 0) {
                    cnt.erase(fruits[j]);
                }
            }
            ans = max(ans, (int)cnt.size());
        }
        return ans;
    }
};